src/checkers/inference/solver/SolverEngine.java

```diff
@@ -113,14 +113,14 @@ public InferenceSolution solve(Map<String, String> configuration, Collection<Slo
      */
     private void configureSolverArgs(final Map<String, String> configuration) {
 
-        final String backEndName = configuration.get(SolverArg.backEndType.name());
+        final String solverName = configuration.get(SolverArg.solver.name());
         final String useGraph = configuration.get(SolverArg.useGraph.name());
         final String solveInParallel = configuration.get(SolverArg.solveInParallel.name());
         final String collectStatistic = configuration.get(SolverArg.collectStatistic.name());
 
-        backEndType = backEndName == null ? BackEndType.MAXSAT : BackEndType.getBackEndType(backEndName);
-        if (backEndType == null) {
-            ErrorReporter.errorAbort("Back end \"" + backEndName + "\" has not been implemented yet.");
+        solverType = solverName == null ? SolverType.MAXSAT : SolverType.getSolverType(solverName);
```

Comment:
I think keep a default solver might be better, as we hope to alleviate arguments burden for users.

CFI would always need a solver to solving constraints, and MAXSAT is a general solver for CFI. So it is a good option to be a default solver.

If a specific type system doesn't support using the default MAXSAT solver, then it could raise that error in it's `XXSolverEngine` class if there were no specific backend is specified (or a better solution might be setting the default solver to another supported one by overriding `sanitizeConfiguration()`.

Reply:
Sounds good.